src/celldega/nbhd/__init__.py

```diff
@@ -205,3 +213,301 @@ def alpha_shape_geojson(gdf_alpha, meta_cluster, inst_alpha):
     geojson_alpha['inst_alpha'] = inst_alpha
 
     return geojson_alpha
+
+def create_hextile(radius, path_landscape_files=None, img_height=100, img_width=100):
+
+    if isinstance(path_landscape_files, str):
+
+        tree = ET.parse(os.path.join(path_landscape_files, "pyramid_images/bound.dzi"))
+        root = tree.getroot()
+        img_width = int(root[0].attrib["Width"])
+        img_height = int(root[0].attrib["Height"])
+
+    hex_width = 2 * radius
+    hex_height = np.sqrt(3) * radius
+    horiz_spacing = 3 / 4 * hex_width  # = 1.5 * r
+    vert_spacing = hex_height
+
+    # Calculate number of hexes
+    n_cols = int(np.ceil(img_width / horiz_spacing)) + 2
+    n_rows = int(np.ceil(img_height / vert_spacing)) + 2
+
+    # Generate hexagons
+    hexagons = []
+
+    for col in range(n_cols):
+        for row in range(n_rows):
+            x = col * horiz_spacing
+            y = row * vert_spacing
+            if col % 2 == 1:
+                y += vert_spacing / 2  # stagger every other column
+
+            # Flat-topped hexagon (aligned horizontally)
+            hexagon = Polygon(
+                [
+                    (
+                        x + radius * np.cos(np.radians(angle)),
+                        y + radius * np.sin(np.radians(angle)),
+                    )
+                    for angle in range(0, 360, 60)
+                ]
+            )
+
+            hexagons.append(hexagon)
+
+    # Define image boundary as a shapely box (left, bottom, right, top)
+    image_bounds = box(0, 0, img_width, img_height)
+
+    # Clip each hexagon to this box
+    clipped_hexes = [
+        hex.intersection(image_bounds)
+        for hex in hexagons
+        if hex.intersects(image_bounds)
+    ]
+
+    # Replace original GeoDataFrame
+    gdf_hextile = gpd.GeoDataFrame(geometry=clipped_hexes)
+
+    if isinstance(path_landscape_files, str):
+
+        gdf_hextile.to_parquet(os.path.join(path_landscape_files, "hextile.parquet"))
+
+    return gdf_hextile
+
+
+def generate_random_points_gdf(n_points=10, x_range=(0, 100), y_range=(0, 100)):
+    points = [
+        Point(random.uniform(*x_range), random.uniform(*y_range))
+        for _ in range(n_points)
+    ]
+    gdf_trx = gpd.GeoDataFrame(geometry=points)
+    return gdf_trx
+
+
+def generate_dummy_image(width=100, height=100):
+
+    data = np.random.randint(0, 256, (height, width, 3), dtype=np.uint8)
+    img = Image.fromarray(data, "RGB")
+    img_np = np.array(img)
+    return img_np
+
+
+def hextile_area(gdf_hextile, tissue_img_shape):
+
+    return round(sum(gdf_hextile.area.tolist())) == round(
+        tissue_img_shape[0] * tissue_img_shape[1]
+    )
+
+
+def hextile_trx_assignment(hextile_assigned_trx):
```

Comment:
can we rename this method to indicate that this is a check on the consistency rather than a assignment calculation